pkg/transport/subscribe/subscribe_data_cache.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022, OpenSergo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscribe
+
+import (
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"sync"
+)
+
+// SubscribeDataCache is a local-cache which cached all the config-data subscribed from opensergo-control-plane.
+type SubscribeDataCache struct {
+	// map[SubscribeKey] SubscribedData
+	cache sync.Map
```

Comment:
Maybe the value type could be `*SubscribedData`? Or we'll suffer from heavy data copy when retrieving the value.

Reply:
changed.